Fix Vue wrapper packaging issue

moores2 opened the pull request:
### Fixes #231 
- Set `main` for package correctly
- Export \src and \dist to package, so that individual components can be imported by clients (like the React wrappers do).

Note: the vue-cli library build generates and extraneous dist/demo.html that does not work. There is an issue for this here: https://github.com/vuejs/vue-cli/issues/3291 

### Review checklist (for reviewers only)
- [ ] Demos all features
- [ ] Documented/annotated
- [ ] Matches UI/UX specs
- [ ] Meets the [code style guide](https://github.com/IBM/carbon-charts/wiki/Code-style-guide)
- [ ] Accessible
- [ ] Mobile first (responsive)
- [ ] RTL support (bidirectional text)
- [ ] Performant (limited bloat)

cal-smith commented on packages/vue/package.json:
```diff
@@ -38,6 +38,10 @@
   },
+  "files": [
+    "/dist",
+    "/src"
```
So, with some changes we made, dist and src aren't automatically part of the build ... you can see this by checking `build.sh` and the ouput by running `npm run build` (we deploy the dist directory) ... if we want to include src (maybe not required?) we'll need to copy it into dist as part of `build.sh`

cal-smith commented on packages/vue/package.json:
```diff
@@ -2,7 +2,7 @@
   "version": "0.11.8",
   "description": "Carbon charting components for Vue",
-  "main": "index.js",
+  "main": "dist/charts-vue.umd.min.js",
```
This should just be 
```suggestion
  "main": "charts-vue.umd.min.js",
```

Since we deploy the contents of the dist directory now

theiliad commented:
@moores2 We need to modify `build.sh` for the `vue` package now to include `src`:
https://github.com/carbon-design-system/carbon-charts/blob/master/packages/vue/build.sh

cal-smith commented:
from my comment here: https://github.com/carbon-design-system/carbon-charts/issues/231#issuecomment-478570856

do we need so ship `src/`? I figure it best to just let consumers deal with the compiled JS ... especially with Vue, consumers really don't get anything by having it available.

theiliad commented:
Yeah it would be preferable if we have a bundle that exports modules separately, and one that includes all where dev can use it through CDNs

moores2 commented:
I have reworked index.js with a simpler approach, and added exports so that the source is not needed in the shipped package. This update should allow clients to load the wrappers in 2 ways: 

Firstly, to just load all of the chart components globally into Vue by using a Vue plug-in, like this:
```
import Vue from 'vue';
import chartsVue from '@carbon/charts-vue';
...
Vue.use(chartsVue);
...
export default {
...
  template:
    '<ccv-bar-chart :data="barData" :options="barOptions"></ccv-bar-chart>'
```
The second option is to allow consumers to selectively import just the wrappers/components that they want, like this:
```
<template>
  ...
  <ccv-bar-chart :data="barData" :options="barOptions"></ccv-bar-chart>
  ...
</template>

<script>
import CcvBarChart from "@carbon/charts-vue";

export default {
  ...
  components: { CcvBarChart },
  ...
};
</script>
```

theiliad commented:
@moores2 @cal-smith This looks good 💯
Do we need any changes in the deploy scripts for the correct bundles/files/directories to be picked up?

cal-smith commented:
Nope we're good to go :+1:

cal-smith commented:
Merging
